src/main/java/org/openpnp/vision/pipeline/stages/Threshold.java

```diff
@@ -45,6 +45,11 @@ public Result process(CvPipeline pipeline) throws Exception {
         Mat mat = pipeline.getWorkingImage();
         int type = invert ? Imgproc.THRESH_BINARY_INV : Imgproc.THRESH_BINARY;
         type |= auto ? Imgproc.THRESH_OTSU : 0;
+	int i,j;
+	i=threshold<-1000?-threshold%100:12;
+	j=threshold<-1000?-threshold/1000:2;
+	if(threshold<0)
+        Imgproc.adaptiveThreshold(mat, mat, 255, Imgproc.ADAPTIVE_THRESH_GAUSSIAN_C, type,i,j); else 
```

Comment:
Can you please explain what this does? Some comments would be helpful. If the purpose is to allow for adaptive threshold this should probably be a new stage.